src/components/Table/DrillDown/components/TableJSX.tsx

```diff
@@ -0,0 +1,265 @@
+          </div>
+        )}
+      </div>
+      <table className="table table-striped table-bordered drill-down-table" {...getTableProps()}>
+        <thead>
+          {headerGroups.map((headerGroup: UseTableHeaderGroupProps<D>) => (
+            <tr {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((c: ColumnInstance<D>, index: number) => {
+                const column = (c as unknown) as ActualColumnInstance<D>;
+                return (
+                  <th
+                    key={`thead-th-${index}`}
+                    // Return an array of prop objects and react-table will merge them appropriately
+                    {...column.getHeaderProps(column.getSortByToggleProps())}
+                  >
+                    {column.render('Header')}
+                    {column.canSort && (
+                      <SortIcon isSorted={column.isSorted} isSortedDesc={column.isSortedDesc} />
+                    )}
+                  </th>
+                );
+              })}
+            </tr>
+          ))}
+        </thead>
+        {data.length > 0 && (
+          <tbody {...getTableBodyProps([{ style: { lineHeight: rHeight } }])}>
+            {page.map((row: Row<D>, idx: number) => {
+              prepareRow(row);
+              return (
+                // Merge user row props in
+                <tr key={`tbody-tr-${idx}`} {...row.getRowProps()}>
+                  {row.cells.map((cell: Cell<D>, i: number) => {
+                    return (
+                      <td
+                        key={`td-${i}`}
+                        // Return an array of prop objects and react-table will merge them appropriately
+                        {...cell.getCellProps([
+                          {
+                            // onclickHandler updates the curentParentId, i.e if the clicked on cell has children
+                            onClick: (e: React.MouseEvent<HTMLElement>) => {
+                              e.stopPropagation();
+                              // onClick will be effective only when drilldingDown and if columnId is the same as linkerField
+                              if (!(props.useDrillDown && cell.column.id === props.linkerField)) {
+                                return;
+                              }
+                              if (props.identifierField && props.parentIdentifierField) {
+                                if (
+                                  props.hasChildren &&
+                                  hasChildrenFunc<D>(cell, props.parentNodes, props.identifierField)
+                                ) {
+                                  const newParentId: string = (row.original as Dictionary)[
+                                    identifierField
+                                  ];
+                                  setCurrentParentId(newParentId);
+                                }
+                              }
+                            },
```

Comment:
@p-netm this looks like drilldown specific functionality in a more general table component, is there ever a case where this component will be used oustide of the drilldown context? If so, would it make sense to pass this in via props?

Reply:
there is a few more good things that react-table provides like sorting and pagination out of the box, these features are required elsewhere in the web ui where the drilldown functionality does not apply. It made sense to me to have a single component that you can configure with boolean flags for what you need per table instance